subiquity/models/filesystem.py

```diff
@@ -1803,10 +1804,18 @@ def can_install(self):
         return (self.is_root_mounted()
                 and not self.needs_bootloader_partition())
 
-    def _should_add_swapfile(self):
-        mount = self._mount_for_path('/')
-        if mount is not None and mount.device.fstype == 'btrfs':
+    def _fstype_supports_swapfile(self, fstype):
+        try:
+            can_use_swapfile('/', fstype)
+            return True
+        except RuntimeError:
```

Comment:
well that's certainly an API contract. do you want to fix it to be less eccentric?

Reply:
https://code.launchpad.net/~dbungert/curtin/+git/curtin/+merge/444944